### DISCUSSION TITLE
Are there plans to implement Python scripting?

### DISCUSSION THREAD

**Discussion by SN-CH** [dail8859/NotepadNext#525] [Ideas]
First and most important of all thank you very much for the work you have already done! I am kind of hybrid user who uses both Linux and Windows and N++ is something I really miss when working in Linux.

While understanding you have neither desire, nor means to make a full-scale clone of N++ that has been in development for... well, I do not know for exactly how long, but for quite a while, I would like to inquire about one specific feature. Are there plans to implement Python scripting for your software? You are probably aware of the N++ add-on called «Python script» and the functionality it provides for users. I certainly realise it can't be a priority for you, but do you consider such a possibility at least in principle? I guess it could be a solution that would allow to circumvent some current limitations of the software for those who are able to script in Python and would become a good replacement for certain functionality that has not been implemented yet.

I have an impression Lua can be used for this purpose, but it is my feeling that the Python user base is considerably larger (just a personal personal opinion).

**Comment by dail8859** [comment]
> Are there plans to implement Python scripting for your software?

No specific plans, but yes I agree (very) long term there needs to be a cross-platform way to script additional functionality in the application. Which means Python is by far the obvious choice. Currently there is a lot of missing functionality, refactoring, etc in the core of the application which would make maintaining some sort of plugin interface burdensome.

> my feeling that the Python user base is considerably larger (just a personal personal opinion).

I agree with this 100%. Lua was never meant to be a long-term solution but was meant more to aid in testing/development. I ported my LuaScript plugin for Notepad++ over to Notepad Next since I needed some way to test things out during runtime. I know Lua is being used for config files but that was more due to ease of use, rather than hoping it to be a long term design decision. I just needed something quick at the time and haven't changed it since.
